pkg/middleware/middleware_test.go

```diff
@@ -66,7 +66,7 @@ func Test_FilterMiddleware(t *testing.T) {
 	// Public, expects to be redirected to the global registry endpoint
```

Comment:
Please add one more test here saying:
```go
// Public, expects to be redirected to the global registry endpoint - test with a trailing slash
res := w.Request("/github.com/gomods/athens/@v/list/").Get()
r.Equal(303, res.Code)
r.Equal(conf.Proxy.GlobalEndpoint+"/github.com/gomods/athens/@v/list/", res.HeaderMap.Get("Location"))
```
difference is in trailing slash in first request